library/waku_thread/waku_thread.nim

```diff
@@ -95,12 +95,16 @@ proc createWakuThread*(): Result[ptr Context, string] =
 
   return ok(ctx)
 
-proc stopWakuNodeThread*(ctx: ptr Context) =
+proc stopWakuNodeThread*(ctx: ptr Context): Result[void, string] =
   running.store(false)
+  let fireRes = ctx.reqSignal.fireSync()
+  if fireRes.isErr():
+    return err(fireRes.error)
   joinThread(ctx.thread)
   discard ctx.reqSignal.close()
   discard ctx.respSignal.close()
   freeShared(ctx)
+  ok()
```

Comment:
nitpic: even though nim supports that I'd rather be more explicit, specially for new Nim devs
```suggestion
  return ok()
```